azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java

```diff
@@ -520,10 +520,15 @@ private void setExpiryFromSAS(AzureSasCredential token) {
         this.expiry = expiryString.map(es -> Instant.parse(es)).orElse(null);
     }
 
+    /**
+     * Return true if this filesystem has SAS credentials with an expiration data attached, and we're within
+     * `buffer` of the expiration. Return false if our credentials don't come with an expiration, or we
+     * aren't within `buffer` of our expiration.
+     */
     public boolean isExpired(Duration buffer) {
         return Optional.ofNullable(this.expiry)
             .map(e -> Instant.now().plus(buffer).isAfter(e))
-            .orElse(true);
+            .orElse(false);
```

Comment:
Reasons `this.expiry` could be `null`:
 * We aren't using a SAS token and thus the credentials will never expire
 * We couldn't parse an expiration date out of the SAS token

In the former case, it's correct to return `false`. In the latter case, the token may actually be expired, but we have no way of knowing, so it seems best to assume it isn't and let things fail and be reopened when it eventually is.